src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,24 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    public Deadline(String desc, String by) {
+        super(desc);
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    @Override
+    public String toDisk() {
+        return String.format("deadline\n%s\n%d\n%s", desc, (done ? 1 : 0), by);
+    }
+}
```

Comment:
Should there be a trailing whitespace here? Also, perhaps it might be good to add a new line to end of file.